stubs/evp_pkey_is_valid_no_ec_key_stub.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <openssl/evp.h>
+
+/**
+ * Helper function for CBMC proofs: checks if EVP_PKEY is valid.
+ * Use this stub when we are *certain* there is no ec_key associated with the key.
+ */
+bool evp_pkey_is_valid(EVP_PKEY *pkey) {
+    return pkey && (pkey->references > 0) && (pkey->ec_key == NULL);
+}
```

Comment:
If we need to be **certain** there is no `ec_key` associated with the key, don't we need an assertion? If we don't, then we need to update the previous stubs, they must have similar behaviors.
```c
/**
 * Checks if EVP_PKEY is valid.
 */
bool evp_pkey_is_valid(EVP_PKEY *pkey) {
    if (!pkey) return false;
    /* We must be sure there is no ec_key associated with the key. */
    assert(pkey->ec_key == NULL);
    return (pkey->references > 0);
}
```

Reply:
Yes, you're right, we should have the assertion here. Updated.